bindings/java/src/main/com/apple/foundationdb/FDBTenant.java

```diff
@@ -198,6 +198,16 @@ public CompletableFuture<Long> verifyBlobRange(byte[] beginKey, byte[] endKey, l
 		}
 	}
 
+	@Override
+	public CompletableFuture<Long> getId(Executor e) {
+		pointerReadLock.lock();
+		try {
+			return new FutureInt64(Tenant_getId(getPtr()), e);
```

Comment:
I'm not sure exactly how this will end up being used, but possibly we could optimize this by only calling out to the native client once and then storing the future in the tenant object to return on subsequent calls.